cmd/xl-storage-disk-id-check.go

```diff
@@ -410,7 +410,13 @@ func (p *xlStorageDiskIDCheck) ListDir(ctx context.Context, volume, dirPath stri
 	}
 	defer done(&err)
 
-	return p.storage.ListDir(ctx, volume, dirPath, count)
+	w := xioutil.NewDeadlineWorker(diskMaxTimeout)
```

Comment:
Bonus change I see. 2 minutes may lead to some cases where overfilled directories time out.

Reply:
Reverted I might have to add this to a layer beneath.